tenacity/__init__.py

```diff
@@ -122,15 +130,20 @@ def call(self, fn, *args, **kwargs):
         self._statistics['attempt_number'] = attempt_number
         self._statistics['idle_for'] = 0
         while True:
-            if self._before_attempts:
-                self._before_attempts(attempt_number)
+            trial_start_time = now()
+            if self.before is not None:
+                self.before(fn, attempt_number)
 
             fut = Future(attempt_number)
             try:
-                fut.set_result(fn(*args, **kwargs))
+                result = fn(*args, **kwargs)
+                trial_end_time = now()
+                fut.set_result(result)
```

Comment:
I think both L140 and L141 should be put in the `else` statement. I was already eager to do it last time, but the more I see thing in this `try` code that shouldn't raise `TryAgain` nor raise any `Exception` embedded, the more I am unhappy.

e.g. I don't want now() failing being a cause of "should I retry?" :-/